src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ToolsApiServiceImpl.scala

```diff
@@ -52,9 +51,10 @@ final case class ToolsApiServiceImpl(authorizationManagementService: Authorizati
           clientAssertionValidator
         ).toFuture
         keyWithClient <- authorizationManagementService.getKeyWithClient(checker.subject, checker.kid)
-        _             <- Future
-          .failed(OrganizationNotAllowed(keyWithClient.client.id))
-          .whenA(requesterId != keyWithClient.client.consumerId)
+        _             <-
+          if (requesterId != keyWithClient.client.consumerId)
+            Future.failed(OrganizationNotAllowed(keyWithClient.client.id))
+          else Future.unit
         _             <- verifyClientAssertionSignature(keyWithClient, checker).toFuture
         _             <- verifyPlatformState(keyWithClient.client, checker).toFuture
```

Comment:
I already discussed privately with @galales about the design of this API. I'm not too fond of the fact that if a client assertion gets inspected for correctness with this API, the API returns an error.

If a `kid` doesn't exist, I expect the token exchange API to return a 400, but when submitting that client assertion to this API, I'll get a 400 too, confusing me about the meaning of that 4xx error. I need to do something different when asking for details about this client assertion, or is the client assertion that I just asked for details a 4xx-producing one?

IMHO, this design might be a bit misleading, I'll prefer having an API that returns a 200 with a description of the possible errors, i.e. a 200 with a `Problem` as a payload.

WDYT @beetlecrunch

Reply:
It's a legit doubt.
I interpreted as "if the validation fails you receive an error", while it returns 200 if the validation succeed.

In the solution where a 200 is returned in any case, how should the frontend be aware (and inform the user) when the validation has been successful and when not?

Reply:
That's a legit doubt too, maybe it's better to summon both @Carminepo2 and @ruggerocastagnola

Reply:
I don't have a clear answer...as @TonioGela suggested probably in this case FE people could help us in this choice.
Independently from this, I was wondering if a cumulative response should be a better way to deal with validations: 200/400 responses could be more useful if return the list of missing validation, but it's a personal taste and I know that this approach impact the auth-server module.